packages/library/src/decode-transaction.ts

```diff
@@ -0,0 +1,75 @@
+import {
+    assertAccountsDecoded,
+    assertAccountsExist,
+    type FetchAccountsConfig,
+    fetchJsonParsedAccounts,
+} from '@solana/accounts';
+import { Address } from '@solana/addresses';
+import type { GetMultipleAccountsApi } from '@solana/rpc-core';
+import type { Rpc } from '@solana/rpc-transport';
+import {
+    type AddressesByLookupTableAddress,
+    type CompilableTransaction,
+    decompileTransaction,
+    getCompiledTransactionDecoder,
+    type ITransactionWithSignatures,
+} from '@solana/transactions';
+
+let compiledTransactionDecoder: ReturnType<typeof getCompiledTransactionDecoder> | undefined = undefined;
+
+type FetchedAddressLookup = {
+    addresses: Address[];
+};
+
+async function fetchLookupTables(
+    lookupTableAddresses: Address[],
+    rpc: Rpc<GetMultipleAccountsApi>,
+    config?: FetchAccountsConfig,
+): Promise<AddressesByLookupTableAddress> {
+    const fetchedLookupTables = await fetchJsonParsedAccounts<FetchedAddressLookup[]>(
```

Comment:
Now that I'm seeing this, all-up, I feel like we're missing the ability to pass an `AbortSignal` here, but I guess that's a limitation of `@solana/codecs-*`. There's no way to pass in per-decode context like that. cc/ @lorisleiva

Reply:
You can pass an abort signal into this function, that `config: FetchAccountsConfig` includes it: https://github.com/solana-labs/solana-web3.js/blob/59e58931e1183c45f7ca084bb939b36906d88c5f/packages/accounts/src/fetch-account.ts#L45-L49

The `decodeTransaction` function takes a config object that's just `FetchAccountsConfig` with an extra optional `lastValidBlockHeight` field, and everything from `FetchAccountsConfig` is passed through to `fetchJsonParsedAccounts`